src/main/java/tool/designpatterns/verifiers/multiclassverifiers/proxy/ProxyFeedbackUtils.java

```diff
@@ -0,0 +1,373 @@
+import static tool.designpatterns.verifiers.multiclassverifiers.proxy.ClassVerification.isSameClassOrInterfaceDeclaration;
+
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+
+import groovy.lang.Tuple2;
+import groovy.lang.Tuple3;
+import tool.designpatterns.Pattern;
+import tool.designpatterns.verifiers.multiclassverifiers.proxy.datahelpers.PartialProxyImplementation;
+import tool.designpatterns.verifiers.multiclassverifiers.proxy.datahelpers.ProxyPatternGroup;
+import tool.feedback.Feedback;
+import tool.feedback.FeedbackTrace;
+
+/**
+ * Utility class to assist with finding issues with / providing feedback for the Proxy pattern
+ * verification.
+ */
+@SuppressWarnings("PMD.CommentSize") // Justification: Did not work when I put it on methods, the
+// reason for why the lines are soo long are due to the number of parameters. There are too many
+// parameters but they would, I believe, take too much to refactor away in a good mannor.
+public final class ProxyFeedbackUtils {
+
+    private ProxyFeedbackUtils() {
+
+    }
+
+    /**
+     * Verifies that all the proxies, interfaces and subjects are used and that if they are only a
+     * part of an 'invalid' proxy pattern, those errors are returned.
+     *
+     * @param proxyPatternGroups the finished proxy groups to check in.
+     * @param proxies            the proxies to check for.
+     * @param interfaces         the interfaces to check for.
+     * @param subjects           the subjects to check for.
+     * @param proxyGroups        the ungrouped proxy/interface groups.
+     * @param subjectGroups      the ungrouped subject/interface groups.
+     *
+     * @return a feedback on how they are used.
+     */
+    public static Feedback allClassesAreUsed(
+        List<ProxyPatternGroup> proxyPatternGroups, List<ClassOrInterfaceDeclaration> proxies,
+        List<ClassOrInterfaceDeclaration> interfaces, List<ClassOrInterfaceDeclaration> subjects,
+        List<PartialProxyImplementation> proxyGroups,
+        List<PartialProxyImplementation> subjectGroups) {
+
+        // Remove the classes used in valid instances from the maps.
+        List<ClassOrInterfaceDeclaration> unusedProxies = getUnused(proxies, proxyPatternGroups,
+            Pattern.PROXY_PROXY);
+        List<ClassOrInterfaceDeclaration> unusedInterfaces = getUnused(interfaces,
+            proxyPatternGroups, Pattern.PROXY_INTERFACE);
+        List<ClassOrInterfaceDeclaration> unusedSubjects = getUnused(subjects, proxyPatternGroups,
+            Pattern.PROXY_SUBJECT);
+
+        List<PartialProxyImplementation> partialProxyGroups = getCorrectPartial(proxyGroups,
+            proxyPatternGroups, Pattern.PROXY_PROXY);
+        List<PartialProxyImplementation> partialSubjectGroups = getCorrectPartial(subjectGroups,
+            proxyPatternGroups, Pattern.PROXY_SUBJECT);
+
+        Tuple2<List<ProxyPatternGroup>, List<ProxyPatternGroup>> validInvalidTuple
```

Comment:
ValidInvalid what?

Reply:
I put a comment on it because I'm not if I can come up with a much better name unfortunately. :P

Comment in 3ea2557